TicketUtil.py

```diff
@@ -119,6 +120,207 @@ def close_requests_session(self):
         """
         self.s.close()
 
+class BugzillaTicket(Ticket):
+    """
+    A BZ Ticket object. Contains BZ-specific methods for working with tickets.
+    """
+
+    def __init__(self, base_url, project_key, ticket_id=None, auth=None, user=None, password=None):
+        self.ticketing_tool = 'Bugzilla'
+
+        # Right now, hardcode auth as 'kerberos', which is the only supported auth for BZ.
+        self.auth = auth if auth else 'kerberos'
+
+        self.credentials = {"login": user, "password": password}
+        self.token = None
+
+        # BZ URLs
+        self.base_url = base_url
+        self.rest_url = '{0}/rest/bug'.format(self.base_url)
+        self.auth_url = '{0}/rest/login'.format(self.base_url)
+
+        # Call our parent class's init method which creates our requests session.
+        super(BugzillaTicket, self).__init__(project_key, ticket_id)
+
+    def craft_ticket_url(self):
+        """
+        Crafts the ticket URL out of the base_url, project_key, and ticket_id.
+        :return: ticket_url: The URL of the ticket.
+        """
+        ticket_url = None
+
+        # If we are receiving a ticket_id, it indicates we'll be doing an update or resolve, so set ticket_url.
+        if self.ticket_id:
+            ticket_url = "{0}/show_bug.cgi?id={1}".format(self.base_url, self.ticket_id)
+
+        return ticket_url
+
+
+    def create_ticket_parameters(self, options_dict):
+        """
+        Creates the payload for the POST request when creating a Bugzilla ticket.
+
+        Example:
+        options_dict = {"product" : "TestProduct",
+                        "component" : "TestComponent",
+                        "version" : "unspecified",
+                        "summary" : "'This is a test bug - please disregard",
+                        "alias" : "SomeAlias",
+                        "op_sys" : "All",
+                        "priority" : "P1",
+                        "rep_platform" : "All"}
+        :param options_dict: Key: Value pairs for updating fields in Bugzilla.
+        :return: params: A dictionary to pass in to the POST request containing ticket details.
+        """
+        # Create our parameters for creating the ticket.
+        params = {'fields': {}}
+        params['fields']['project'] = {'key': self.project_key}
+
+        params = {"product": str(self.project_key)}
```

Comment:
In line 179, you're overwriting the params dictionary you created in lines 176 and 177. If the code is currently working, that means that line 179 is the correct syntax, meaning you can delete lines 176-177.